### ISSUE TITLE
Support for inline services

### ISSUE THREAD

**Issue by rejinka** [mnapoli/fluent-symfony#5]
In the XML-Loader, we have support for sth like this:

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<container xmlns="http://symfony.com/schema/dic/services"
           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
           xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">

    <services>
        <service id="foo" class="Foo">
            <argument type="service">
                <service class="Bar" />
            </argument>
        </service>
    </services>
</container>
```

So the Foo gets injected an instance of Bar, without having the possibility to fetch or reference this instance from the container. It is only available for this one service. I'd really like to see something like

```php
return [
    create(Foo::class)
        ->arguments(create(Bar::class)), // ->arguments(inline(Bar::class))?
];
```

**Comment by rejinka** [comment]
With the current architecture, i have some trouble implementing this feature. Instead of `CreateDefinitionHelper::register()` i'd rather need sth. like `CreateDefinitionHelper::getDefinition()` as the `Definition` does not directly gets registered in the container but as part of the actual service definition. I need to fetch the definition directly, as `CreateDefinitionHelper::arguments()` and `CreateDefinitionHelper::method()` need an instance of `Definition`  **somehow**. 

I think it would be necessary to use the Builder pattern and move the registration logic away from the Helper to `PhpConfigLoader`. This would make the loader much more complicated, as it would need to know, how to handle the Helper classes (handler classes?). Furthermore `CreateDefinitionHelper::arguments($arguments)` needed to iterate all `$arguments` and replace each `CreateDefinitionHelper` with `CreateDefinitionHelper::getDefinition()`.

Could use some thoughts here.

**Comment by mnapoli** [comment]
Right I see, not easy indeed :/

Also because of [`ExtensionConfiguration`](https://github.com/mnapoli/fluent-symfony/blob/master/src/DefinitionHelper/ExtensionConfiguration.php) (which allows to define configuration for some extensions) simply changing `register()` into `getDefinition()` is not possible.

Here is an idea (haven't thought too much about it though), having 2 interfaces:

- `DefinitionHelper` (with a `getDefinition()` method), implemented in `create()`, `autowire()`, `factory()`
- `Registrable` (with a `register()` method), implemented in `extension()`, `alias()` as well as `create()`, `autowire()`, `factory()` (which would simply register the definition)

`Registrable` is a crappy name, hopefully we can find better…

**Comment by rejinka** [comment]
Sounds a bit fragile to me, but we'll know, as soon as i tried implementing this. Then we can discuss improvements and perhaps another approach :-)

**Comment by mnapoli** [comment]
Yeah I tried to implement some of these ideas and it was a fail, so let's discuss some actual code it will be more productive I think :)
